galpy/potential_src/SpiralArmsPotential.py

```diff
@@ -0,0 +1,584 @@
+###############################################################################
+#  SpiralArmsPotential.py: class that implements the spiral arms potential
+#                           from Cox and Gomez (2002)
+#
+#  https://arxiv.org/abs/astro-ph/0207635v1
+#
+#  Phi(r, phi, z) = -4*pi*G*H*rho0*exp(-(r-r0)/Rs)*sum(Cn/(Kn*Dn)*cos(n*gamma)*sech(Kn*z/Bn)^Bn)
+#  NOTE: Methods do not take array inputs.
+###############################################################################
+
+from __future__ import division
+from functools import wraps
+from galpy.potential_src.Potential import Potential, _APY_LOADED
+from galpy.util import bovy_conversion
+import numpy as np
+
+if _APY_LOADED:
+    from astropy import units
+
+
+def check_inputs_not_arrays(func):
+    """
+    Decorator to check inputs and throw TypeError if any of the inputs are arrays.
+    Methods potentially return with silent errors if inputs are not checked.
+    """
+    @wraps(func)
+    def func_wrapper(self, R, z, phi, t):
+        if hasattr(R, '__len__') or hasattr(z, '__len__') or hasattr(phi, '__len__') or hasattr(t, '__len__'):
+            raise TypeError('Methods in SpiralArmsPotential do not accept array inputs. Please input scalars.')
```

Comment:
We need to add a test of this Error. Write a little test function in ``test_potential.py`` similar to [this function](https://github.com/jobovy/galpy/blob/master/tests/test_potential.py#L1626) for that.